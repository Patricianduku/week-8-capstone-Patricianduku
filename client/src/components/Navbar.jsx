import React, { useState } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { Menu, X, Heart, Home, BookOpen, Users, Phone, LogIn, Settings, LogOut, User } from 'lucide-react';
import tulizaLogo from '../assets/tuliza-logo.png';
import DarkModeToggle from './DarkModeToggle';
import { useTranslation } from 'react-i18next';
import { useAuth } from '../hooks/useAuth';

const Navbar = () => {
  const [isOpen, setIsOpen] = useState(false);
  const location = useLocation();
  const { i18n, t } = useTranslation();
  const { user, logout } = useAuth();

  const navigation = [
    { name: t('Home'), href: '/', icon: Home },
    { name: t('Journal'), href: '/journal', icon: BookOpen },
    { name: t('Support Rooms'), href: '/support', icon: Users },
    { name: t('Resources'), href: '/resources', icon: Phone },
    { name: t('Settings'), href: '/settings', icon: Settings },
    { name: t('Pricing'), href: '/pricing', icon: Heart },
  ];

  const isActive = (path) => location.pathname === path;

  const handleLanguageChange = (e) => {
    i18n.changeLanguage(e.target.value);
  };

  return (
    <nav className="w-full bg-white dark:bg-gray-950 shadow-md sticky top-0 z-50">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 flex items-center justify-between h-20">
        <Link to="/" className="flex items-center gap-2">
          <img src={tulizaLogo} alt={t('Tuliza Logo')} className="h-16 w-auto" />
          <span className="text-2xl font-bold text-primary-700 dark:text-primary-200 tracking-wide">{t('Tuliza')}</span>
        </Link>
        {/* Desktop Navigation */}
        <div className="hidden md:flex items-center space-x-4">
          {navigation.map((item) => {
            const Icon = item.icon;
            return (
              <Link
                key={item.name}
                to={item.href}
                className={`flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 ${
                  isActive(item.href)
                    ? 'text-primary-600 bg-primary-50 dark:text-primary-400 dark:bg-primary-900/20'
                    : 'text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700'
                }`}
              >
                <Icon className="h-4 w-4" />
                <span>{item.name}</span>
              </Link>
            );
          })}
          <DarkModeToggle />
          {/* Language Switcher */}
          <select
            value={i18n.language}
            onChange={handleLanguageChange}
            className="rounded-lg px-2 py-1 text-gray-900 focus:outline-none focus:ring-2 focus:ring-primary-400"
            aria-label={t('Language')}
          >
            <option value="en">{t('English')}</option>
            <option value="sw">{t('Swahili')}</option>
          </select>
          {user ? (
            <div className="flex items-center space-x-2">
              <User className="h-5 w-5 text-primary-600 dark:text-primary-400" />
              <span className="font-semibold text-primary-700 dark:text-primary-200">{user.name}</span>
              <button
                onClick={logout}
                className="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium text-red-600 hover:bg-red-50 dark:hover:bg-red-900/20 transition-colors"
              >
                <LogOut className="h-4 w-4" />
                <span>{t('Logout')}</span>
              </button>
            </div>
          ) : (
            <Link
              to="/login"
              className="flex items-center space-x-1 btn-primary"
            >
              <LogIn className="h-4 w-4" />
              <span>{t('Login')}</span>
            </Link>
          )}
        </div>
        {/* Mobile menu button */}
        <div className="md:hidden flex items-center">
          <div className="mr-2">
            <DarkModeToggle />
          </div>
          {/* Language Switcher for mobile - REMOVED FROM HERE */}
          <button
            onClick={() => setIsOpen(!isOpen)}
            className="text-gray-600 hover:text-gray-900 dark:text-gray-300 dark:hover:text-gray-100 focus:outline-none focus:text-gray-900 dark:focus:text-gray-100 p-2 transition-colors duration-200"
          >
            {isOpen ? <X className="h-6 w-6" /> : <Menu className="h-6 w-6" />}
          </button>
        </div>
      </div>
      {/* Mobile Navigation */}
      {isOpen && (
        <div className="md:hidden bg-white dark:bg-gray-800 border-t border-gray-200 dark:border-gray-700 transition-colors duration-200">
          <div className="px-2 pt-2 pb-3 space-y-1">
            {navigation.map((item) => {
              const Icon = item.icon;
              return (
                <Link
                  key={item.name}
                  to={item.href}
                  onClick={() => setIsOpen(false)}
                  className={`flex items-center space-x-2 px-3 py-2 rounded-md text-base font-medium transition-colors duration-200 ${
                    isActive(item.href)
                      ? 'text-primary-600 bg-primary-50 dark:text-primary-400 dark:bg-primary-900/20'
                      : 'text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700'
                  }`}
                >
                  <Icon className="h-5 w-5" />
                  <span>{item.name}</span>
                </Link>
              );
            })}
            {user ? (
              <div className="flex items-center space-x-2 px-3 py-2">
                <User className="h-5 w-5 text-primary-600 dark:text-primary-400" />
                <span className="font-semibold text-primary-700 dark:text-primary-200">{user.name}</span>
                <button
                  onClick={() => { setIsOpen(false); logout(); }}
                  className="flex items-center space-x-1 px-3 py-2 rounded-md text-base font-medium text-red-600 hover:bg-red-50 dark:hover:bg-red-900/20 transition-colors"
                >
                  <LogOut className="h-5 w-5" />
                  <span>{t('Logout')}</span>
                </button>
              </div>
            ) : (
              <Link
                to="/login"
                onClick={() => setIsOpen(false)}
                className="flex items-center space-x-2 px-3 py-2 text-base font-medium text-primary-600 hover:bg-primary-50 dark:text-primary-400 dark:hover:bg-primary-900/20 rounded-md transition-colors duration-200"
              >
                <LogIn className="h-5 w-5" />
                <span>{t('Login')}</span>
              </Link>
            )}
            {/* Language Switcher for mobile - MOVED HERE */}
            <div className="px-3 pt-2">
              <select
                value={i18n.language}
                onChange={handleLanguageChange}
                className="rounded-lg px-2 py-1 text-gray-900 w-full focus:outline-none focus:ring-2 focus:ring-primary-400"
                aria-label={t('Language')}
              >
                <option value="en">{t('English')}</option>
                <option value="sw">{t('Swahili')}</option>
              </select>
            </div>
          </div>
        </div>
      )}
    </nav>
  );
};

export default Navbar;