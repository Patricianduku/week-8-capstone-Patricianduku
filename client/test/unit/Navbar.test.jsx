import { render, screen, fireEvent } from '@testing-library/react';
import Navbar from '../../src/components/Navbar';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';

beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value:vi.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: vi.fn(), // deprecated
        removeListener:vi.fn(), // deprecated
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
        dispatchEvent: vi.fn(),
      })),
    });
  });
  vi.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: (key) => key,
        i18n: {
          language: 'en',
          changeLanguage: vi.fn(),
        },
    }),
  }));
    
const renderWithRouter = (ui) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>);
};

describe('Navbar', () => {
  it('renders all navigation links', () => {
    renderWithRouter(<Navbar />);

    expect(screen.getByText(/Home/i)).toBeInTheDocument();
    expect(screen.getByText(/Support Rooms/i)).toBeInTheDocument();
    expect(screen.getByText(/Login/i)).toBeInTheDocument();
  });

  it('toggles dark mode when button is clicked', () => {
    renderWithRouter(<Navbar />);
    const toggle = screen.getByRole('button', { name: /toggle dark mode/i });

    fireEvent.click(toggle);

    // Add expectations depending on your logic (e.g., class change)
    // This line is just a placeholder
    expect(document.documentElement.classList.contains('dark')).toBe(true);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
