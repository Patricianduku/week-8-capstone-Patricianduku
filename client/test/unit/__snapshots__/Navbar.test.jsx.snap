// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="w-full bg-white dark:bg-gray-950 shadow-md sticky top-0 z-50"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 flex items-center justify-between h-20"
    >
      <a
        class="flex items-center gap-2"
        data-discover="true"
        href="/"
      >
        <img
          alt="Tuliza Logo"
          class="h-16 w-auto"
          src="/src/assets/tuliza-logo.png"
        />
        <span
          class="text-2xl font-bold text-primary-700 dark:text-primary-200 tracking-wide"
        >
          Tuliza
        </span>
      </a>
      <div
        class="hidden md:flex items-center space-x-4"
      >
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-primary-600 bg-primary-50 dark:text-primary-400 dark:bg-primary-900/20"
          data-discover="true"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-house h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
            />
            <path
              d="M3 10a2 2 0 0 1 .709-1.528l7-5.999a2 2 0 0 1 2.582 0l7 5.999A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700"
          data-discover="true"
          href="/journal"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-book-open h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 7v14"
            />
            <path
              d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
            />
          </svg>
          <span>
            Journal
          </span>
        </a>
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700"
          data-discover="true"
          href="/support"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-users h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <path
              d="M16 3.128a4 4 0 0 1 0 7.744"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
          </svg>
          <span>
            Support Rooms
          </span>
        </a>
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700"
          data-discover="true"
          href="/resources"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-phone h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.832 16.568a1 1 0 0 0 1.213-.303l.355-.465A2 2 0 0 1 17 15h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2A18 18 0 0 1 2 4a2 2 0 0 1 2-2h3a2 2 0 0 1 2 2v3a2 2 0 0 1-.8 1.6l-.468.351a1 1 0 0 0-.292 1.233 14 14 0 0 0 6.392 6.384"
            />
          </svg>
          <span>
            Resources
          </span>
        </a>
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700"
          data-discover="true"
          href="/settings"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-settings h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
          <span>
            Settings
          </span>
        </a>
        <a
          class="flex items-center space-x-1 px-3 py-2 rounded-md text-sm font-medium transition-colors duration-200 text-gray-600 hover:text-primary-600 hover:bg-gray-50 dark:text-gray-300 dark:hover:text-primary-400 dark:hover:bg-gray-700"
          data-discover="true"
          href="/pricing"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-heart h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
            />
          </svg>
          <span>
            Pricing
          </span>
        </a>
        <button
          aria-label="Switch to dark mode"
          class="p-2 rounded-lg bg-gray-200 hover:bg-gray-300 dark:bg-gray-700 dark:hover:bg-gray-600 transition-colors duration-200 focus:outline-none focus:ring-2 focus:ring-primary-500 focus:ring-offset-2 dark:focus:ring-offset-gray-800"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-moon h-5 w-5 text-gray-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
        </button>
        <select
          aria-label="Language"
          class="rounded-lg px-2 py-1 text-gray-900 focus:outline-none focus:ring-2 focus:ring-primary-400"
        >
          <option
            value="en"
          >
            English
          </option>
          <option
            value="sw"
          >
            Swahili
          </option>
        </select>
        <a
          class="flex items-center space-x-1 btn-primary"
          data-discover="true"
          href="/login"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-log-in h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m10 17 5-5-5-5"
            />
            <path
              d="M15 12H3"
            />
            <path
              d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
            />
          </svg>
          <span>
            Login
          </span>
        </a>
      </div>
      <div
        class="md:hidden flex items-center"
      >
        <div
          class="mr-2"
        >
          <button
            aria-label="Switch to dark mode"
            class="p-2 rounded-lg bg-gray-200 hover:bg-gray-300 dark:bg-gray-700 dark:hover:bg-gray-600 transition-colors duration-200 focus:outline-none focus:ring-2 focus:ring-primary-500 focus:ring-offset-2 dark:focus:ring-offset-gray-800"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-moon h-5 w-5 text-gray-600"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
          </button>
        </div>
        <button
          class="text-gray-600 hover:text-gray-900 dark:text-gray-300 dark:hover:text-gray-100 focus:outline-none focus:text-gray-900 dark:focus:text-gray-100 p-2 transition-colors duration-200"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-menu h-6 w-6"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 12h16"
            />
            <path
              d="M4 18h16"
            />
            <path
              d="M4 6h16"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
